### PR TITLE
feat: support to create subscription

### LINKED ISSUE #323: support to create subscription
### Is your feature request related to a problem? Please describe.（如果该特性或者功能和已知某个问题相关，请描述问题是什么）

support `create subscription`:

```sql
CREATE SUBSCRIPTION "<subscription_name>" ON "<db_name>"."<retention_policy>" 
DESTINATIONS <ALL|ANY> "<subscription_endpoint_host>"
```

### Describe the solution you'd like（问题的解决办法是什么）

1. The `ts-sql` receives the creation request.
2. The `ts-meta` persistences the subscription.

### Describe alternatives you've considered（描述您考虑过的替代方案）

_No response_

### Additional context（其他补充内容）

_No response_

### PR DESCRIPTION
**Pull request by ZelinMa557**
### What problem does this PR solve?

Issue Number: close #323

### What is changed and how it works?
Add functions to `meta_client`, and enable create subscription statement in `statement_executor` 

### How Has This Been Tested?
Add unit test to `data_test.go` & `meta_client_test.go`
image
image
Also I add integretion test:
image

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/metaclient/meta_client.go [hunk 1]
```diff
@@ -2002,8 +2002,24 @@ func (c *Client) DeleteMetaNode(id uint64) error {
...
+}
+
 // CreateSubscription creates a subscription against the given database and retention policy.
 func (c *Client) CreateSubscription(database, rp, name, mode string, destinations []string) error {
+	for _, destination := range destinations {
+		if err := pingServer(destination); err != nil {
+			return errors.New("Fail to ping " + destination)
+		}
+	}
 	return c.retryUntilExec(proto2.Command_CreateSubscriptionCommand, proto2.E_CreateSubscriptionCommand_Command,
```
Comment (shilinlee) [inline lib/metaclient/meta_client.go:2023]: can `retryUntilExec` replace with `retryExec`? We can discuss it.

File: lib/metaclient/meta_client.go [hunk 2]
```diff
@@ -2002,8 +2002,24 @@ func (c *Client) DeleteMetaNode(id uint64) error {
...
+		return err
+	}
+	return nil
+}
+
 // CreateSubscription creates a subscription against the given database and retention policy.
 func (c *Client) CreateSubscription(database, rp, name, mode string, destinations []string) error {
+	for _, destination := range destinations {
+		if err := pingServer(destination); err != nil {
+			return errors.New("Fail to ping " + destination)
```
Comment (shilinlee) [inline lib/metaclient/meta_client.go:2020]:
```suggestion
			return fmt.Errorf("Fail to ping %s", destination)
```

File: open_src/influx/meta/data.go [hunk 3]
```diff
@@ -1760,14 +1760,14 @@ func (data *Data) pruneShardGroups(id uint64) error {
...
-// validateURL returns an error if the URL does not have a port or uses a scheme other than UDP or HTTP.
+// validateURL returns an error if the URL does not have a port or uses a scheme other than HTTP.
 func validateURL(input string) error {
 	u, err := url.Parse(input)
 	if err != nil {
 		return ErrInvalidSubscriptionURL(input)
 	}
 
-	if u.Scheme != "udp" && u.Scheme != "http" && u.Scheme != "https" {
+	if u.Scheme != "http" && u.Scheme != "https" {
```
Comment (shilinlee) [inline open_src/influx/meta/data.go:1770]: Why not support `udp`?

**Review by shilinlee — CHANGES_REQUESTED** [review]

File: lib/metaclient/meta_client.go (hunk 1 above)
Comment (ZelinMa557) [reply lib/metaclient/meta_client.go:2023]: I compared `retryUntilExec` and `retryExec`, `retryUntilExec` is a simple warp of `retryExec`, but `retryUntilExec` wait for local updates (of metadata) to catch up with remote updates. In `showSubscription`, client will only use local cached data, if we use `retryExec`, maybe it sometimes leads to inconsistencies?

File: lib/metaclient/meta_client.go (hunk 2 above)
Comment (ZelinMa557) [reply lib/metaclient/meta_client.go:2020]: OK, this is a good idea, I will change it.

File: open_src/influx/meta/data.go (hunk 3 above)
Comment (ZelinMa557) [reply open_src/influx/meta/data.go:1770]: I don't know if openGemini support `UDP` write or not

Comment (shilinlee) [reply open_src/influx/meta/data.go:1770]: Maybe openGemini should support `UDP` protocol in the near future.

**Review by shilinlee — APPROVED** [review]

**Pull request merged**